fastcrypto-vdf/src/class_group/discriminant.rs

```diff
@@ -1,28 +1,27 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{math::hash_prime, Parameter, ToBytes};
+use crate::math::hash_prime;
+use crate::math::parameterized_group::Parameter;
 use fastcrypto::error::FastCryptoError::InvalidInput;
 use fastcrypto::error::{FastCryptoError, FastCryptoResult};
-use num_bigint::{BigInt, Sign, ToBigInt};
+use num_bigint::{BigInt, ToBigInt};
 use num_integer::Integer;
 use num_traits::Signed;
+use serde::Deserializer;
+use serde::{Deserialize, Serialize};
 use std::ops::Neg;
 
-/// A discriminant for an imaginary class group. The discriminant is a negative integer which is
-/// equal to 1 mod 8.
-#[derive(PartialEq, Eq, Debug, Clone)]
-pub struct Discriminant(BigInt);
-
-impl ToBytes for Discriminant {
-    fn to_bytes(&self) -> Vec<u8> {
-        self.0.to_bytes_be().1
-    }
-}
+/// A discriminant for an imaginary class group. The discriminant is a negative integer congruent to
+/// 1 mod 8.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize)]
+pub struct Discriminant(#[serde(with = "crate::class_group::bigint_serde")] BigInt);
 
 impl TryFrom<BigInt> for Discriminant {
     type Error = FastCryptoError;
 
+    /// A valid discriminant should be a negative prime congruent to 1 mod 8. The primality is _not_
+    /// checked.
```

Comment:
Is there a way for a quick heuristic deterministic primality check?

Reply:
It will be too slow for large discriminants. Alternatively, we can add a `from`/`from_unchecked` method which doesn't check primality and test the primality in the `try_from` method.

Reply:
I've added a `check_primality` function instead.